Source/ServerTrustEvaluation.swift

```diff
@@ -104,17 +121,18 @@ public final class DefaultTrustEvaluator: ServerTrustEvaluating {
         self.validateHost = validateHost
     }
 
-    /// Evaluates the given `SecTrust` value for the given `host`.
-    ///
-    /// - Parameters:
-    ///   - trust: The `SecTrust` value to evaluate.
-    ///   - host:  The host for which to evaluate the `SecTrust` value.
-    /// - Returns: Whether or not the evaluator considers the `SecTrust` value valid for `host`.
-    public func evaluate(_ trust: SecTrust, forHost host: String) -> Bool {
-        let policy = SecPolicyCreateSSL(true, validateHost ? host as CFString : nil)
-        SecTrustSetPolicies(trust, policy)
+    public func evaluate(_ trust: SecTrust, forHost host: String) throws -> Bool {
+        try trust.validate(policy: .default) { (status, result) in
+            AFError.serverTrustEvaluationFailed(reason: .defaultEvaluationFailed(output: .init(host, trust, status, result)))
+        }
+
+        if validateHost {
+            try trust.validate(policy: .hostname(host)) { (status, result) in
+                AFError.serverTrustEvaluationFailed(reason: .hostValidationFailed(output: .init(host, trust, status, result)))
+            }
+        }
 
-        return trust.isValid
+        return true
```

Comment:
This logic might point out a small design flaw in the APIs. The `evaluate` method does not gather the `Bool` result anywhere from either `validate` call. It just returns `true` if it doesn't encounter an error. This seems to indicate that the APIs should probably just throw, not also return a `Bool`. Is there a specific reason to keep both? Potentially clients adding their own evaluations and choosing to just return `true` or `false`? I can't think of a great reason off the top of my head to keep that. If it's convenience for the test suite, we should be able to wrap the calls in the tests to offer a Bool option.

Reply:
You're right, the API feels better without the return value.